sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -49,24 +49,32 @@ static int _add_addr_to_entry(ng_ipv6_netif_t *entry, const ng_ipv6_addr_t *addr
                                       sizeof(addr_str)),
                   prefix_len, entry->pid);
 
-            if (anycast || ng_ipv6_addr_is_multicast(addr)) {
-                if (ng_ipv6_addr_is_multicast(addr)) {
-                    entry->addrs[i].prefix_len = NG_IPV6_ADDR_BIT_LEN;
-                }
+            entry->addrs[i].prefix_len = prefix_len;
 
-                entry->addrs[i].flags = NG_IPV6_NETIF_ADDR_FLAGS_NON_UNICAST;
+            if ((flags & NG_IPV6_NETIF_ADDR_FLAGS_NON_UNICAST) ||
+                ng_ipv6_addr_is_multicast(addr)) {
+                entry->addrs[i].flags |= NG_IPV6_NETIF_ADDR_FLAGS_NON_UNICAST;
             }
             else {
-                entry->addrs[i].prefix_len = prefix_len;
+                if (!ng_ipv6_addr_is_link_local(addr)) {
+                    /* add also corresponding link-local address */
+                    ng_ipv6_addr_t ll_addr;
+
+                    ll_addr.u64[1] = addr->u64[1];
+                    ng_ipv6_addr_set_link_local_prefix(&ll_addr);
 
-                entry->addrs[i].flags = NG_IPV6_NETIF_ADDR_FLAGS_UNICAST;
+                    entry->addrs[i].flags = NG_IPV6_NETIF_ADDR_FLAGS_UNICAST;
+
+                    return _add_addr_to_entry(entry, &ll_addr, 64,
+                                              flags | NG_IPV6_NETIF_ADDR_FLAGS_NDP_ON_LINK);
```

Comment:
Do I get this right: the corresponding link-local address is added, too, and the pointer to this second address is returned? Is this sensible?